impl/src/main/java/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImpl.java

```diff
@@ -0,0 +1,262 @@
+            } else if (pemContent instanceof PrivateKeyInfo) {
+                PrivateKeyInfo privateKeyInfo = (PrivateKeyInfo) pemContent;
+                privateKey = jcaPEMKeyConverter.getPrivateKey(privateKeyInfo);
+            } else {
+                throw new IllegalArgumentException("Unsupported Private Key format '" +
+                    pemContent.getClass().getSimpleName() + '"');
+            }
+        }
+
+        return privateKey;
+    }
+
+    /**
+     * Create token client config from the supplied API client config.
+     *
+     * Token client needs to retry http 401 errors only once which is not the case with the API client.
+     * We therefore effect this token client specific config by setting 'retryMaxElapsed'
+     * & 'retryMaxAttempts' fields.
+     *
+     * @param apiClientConfiguration
+     * @return ClientConfiguration to be used by token retrieval client.
+     */
+    ClientConfiguration constructTokenClientConfig(ClientConfiguration apiClientConfiguration) {
+        ClientConfiguration tokenClientConfiguration = new ClientConfiguration();
+
+        if (apiClientConfiguration.getClientCredentialsResolver() != null)
+            tokenClientConfiguration.setClientCredentialsResolver(apiClientConfiguration.getClientCredentialsResolver());
+
+        if (apiClientConfiguration.isCacheManagerEnabled())
+            tokenClientConfiguration.setCacheManagerEnabled(apiClientConfiguration.isCacheManagerEnabled());
+
+        tokenClientConfiguration.setCacheManagerTti(apiClientConfiguration.getCacheManagerTti());
+        tokenClientConfiguration.setCacheManagerTtl(apiClientConfiguration.getCacheManagerTtl());
+
+        if (apiClientConfiguration.getCacheManagerCaches() != null)
+            tokenClientConfiguration.setCacheManagerCaches(apiClientConfiguration.getCacheManagerCaches());
+
+        if (apiClientConfiguration.getRequestAuthenticatorFactory() != null)
+            tokenClientConfiguration.setRequestAuthenticatorFactory(apiClientConfiguration.getRequestAuthenticatorFactory());
+
+        if (apiClientConfiguration.getBaseUrlResolver() != null)
+            tokenClientConfiguration.setBaseUrlResolver(apiClientConfiguration.getBaseUrlResolver());
+
+        tokenClientConfiguration.setAuthenticationScheme(AuthenticationScheme.OAUTH2);
+        tokenClientConfiguration.setAuthorizationMode(AuthorizationMode.PRIVATE_KEY);
+        tokenClientConfiguration.setClientId(apiClientConfiguration.getClientId());
+        tokenClientConfiguration.setScopes(apiClientConfiguration.getScopes());
+        tokenClientConfiguration.setPrivateKey(apiClientConfiguration.getPrivateKey());
+
+        // setting this to '0' will disable this check and only 'retryMaxAttempts' will be effective
+        tokenClientConfiguration.setRetryMaxElapsed(0);
+        // retry only once for token requests (http 401 errors)
+        tokenClientConfiguration.setRetryMaxAttempts(1);
+
+        return tokenClientConfiguration;
```

Comment:
Looks like the proxy configuration is not copied from the original

Reply:
fixed